src/crl.rs

```diff
@@ -67,7 +135,28 @@ impl<'a> TryFrom<&'a [u8]> for CertRevocationList<'a> {
     ///
     /// [^1]: <https://www.rfc-editor.org/rfc/rfc5280#section-5>
     fn try_from(crl_der: &'a [u8]) -> Result<Self, Self::Error> {
-        parse_crl(untrusted::Input::from(crl_der))
+        // Try to parse the CRL.
+        let crl = parse_crl(untrusted::Input::from(crl_der))?;
+
+        // Iterate through the revoked certificate entries to ensure they are valid so we can
+        // yield an error up-front instead of on first iteration by the caller.
+        // TODO(@cpu): Should we do this up-front, or accept late discovery of CRL entry errs?
+        for cert_result in crl.into_iter() {
+            cert_result?;
+        }
+
+        Ok(crl)
+    }
+}
+
+impl<'a> CertRevocationList<'a> {
+    /// Try to find a [`RevokedCert`] in the CRL that has a serial number matching `serial`. This
+    /// method will ignore any [`RevokedCert`] entries that do not parse successfully. To handle
+    /// parse errors use [`CertRevocationList`]'s [`IntoIterator`] trait.
+    pub fn find_serial(&self, serial: &[u8]) -> Option<RevokedCert> {
+        self.into_iter()
+            .filter_map(|parse_res| parse_res.ok())
+            .find(|revoked_cert| revoked_cert.serial_number.eq(serial))
     }
```

Comment:
Yup!

Reply:
I filed an explicit issue for this so it can be referenced with a comment and not lost to the sands of time: https://github.com/rustls/webpki/issues/80